### DISCUSSION TITLE
Administration user en attente de validation

### DISCUSSION THREAD

**Discussion by Jireck-npds** [npds/npds_dune#558] [Ideas]
il y a un soucis de comptage ou 

<img width="593" alt="image" src="https://github.com/npds/npds_dune/assets/16939826/7d42cecf-cf73-4b64-b466-d21b724f7ed4">

**Comment by jipexu** [comment]
???? tu as du faire un mauvais insert dans ta table ....
dans users_status la colonne open (tes valeurs sont inversées ? ) 0 au lieu de 1 ? pour les utilisateurs autorisés ....???? ou des données précédentes gardées en mémoires ...

**Comment by Jireck-npds** [comment]
non je n'ai pas fait de mauvaise insert.
dans le menu tu as 5242 users en attente. ils correspondent d'après toi au critère open qui est la coche qui autorise la connexion de l'utilisateur.
Jusque là pas de soucis.
quand je clique sur l'icone du menu, cela m'affiche 126 utilisateurs. pourquoi pas 5242 ? 
ensuite il affiche ces 126 users dont voici une entree et les entetes
<img width="1032" alt="image" src="https://github.com/npds/npds_dune/assets/16939826/50c86682-b327-4c9a-a7ee-bc3e6920cbda">
A quoi corresponds la date dans l'image ?

**Comment by Jireck-npds** [comment]
ok je viens de regarder les requêtes et c'est normal que ca n'affiche pas la même chose et que c'est entre guillemet faux.

Le premier se base sur les entrées users_status uniquement alors que le deuxième fait la bonne requête sur la table user et la table users_status

**Comment by jipexu** [comment]
la date est la date d'enregistrement de l'utilisateur ...
tes tables sont synchronisées ???
parce qu'a première vue les requêtes me semblent correctes et je n'observe pas de dysfonctionnement sur le labo...
<img width="841" alt="Capture d’écran 2024-05-15 à 18 10 29" src="https://github.com/npds/npds_dune/assets/328101/00881160-0c80-4285-954e-d554ed44a23e">


- dans la première on prend tous les utilisateur qui ne sont pas autorisés  ...(pas besoin des données de la jointure donc seulement dans la table users_status)
- dans la deuxième comme on a besoin de plus de données (alors jointure des deux tables ...) sur le uid ...
- ???

**Comment by Jireck-npds** [comment]
Oui j'ai épuré la table users car il n'y a pas de suppression integré des utilisateurs.
maintenant faut que je trouve un requete pour epuré la table users_status et user_extends.

L'ideal c'est d'avoir la date de création et la date de derniere connexion.
car j'épure regulierement les users "morts" car non connecté depuis 2ans et qui ont soit recréer un compte ou pas

**Comment by Jireck-npds** [comment]
donc au moi dire ce que c'est la date

**Comment by jipexu** [comment]
> **la date est la date d'enregistrement de l'utilisateur ...** tes tables sont synchronisées ??? parce qu'a première vue les requêtes me semblent correctes et je n'observe pas de dysfonctionnement sur le labo... <img alt="Capture d’écran 2024-05-15 à 18 10 29" width="841" src="https://private-user-images.githubusercontent.com/328101/330882061-00881160-0c80-4285-954e-d554ed44a23e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU5Mjg5NDgsIm5iZiI6MTcxNTkyODY0OCwicGF0aCI6Ii8zMjgxMDEvMzMwODgyMDYxLTAwODgxMTYwLTBjODAtNDI4NS05NTRlLWQ1NTRlZDQ0YTIzZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxN1QwNjUwNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZjA5MjZmZTZlMzlkZGU4NWIyZmFlNDFlMDU2NDg1ZjMzM2JiY2QxNTkwZmZmMWEzN2M0YjZkNjNjMjI2ODY5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.9apOUdK6FpF0TIRGxesiJiZ6-e2Q4yqz2D-v-rylu7Q">
> 
> * dans la première on prend tous les utilisateur qui ne sont pas autorisés  ...(pas besoin des données de la jointure donc seulement dans la table users_status)
> * dans la deuxième comme on a besoin de plus de données (alors jointure des deux tables ...) sur le uid ...
> * ???

**Comment by jipexu** [comment]
regarde    case 'delUserConf': dans admin/users.php 
"il te suffit" de modifier la requête de selection (avec un critère sur la date de dernière connection user_lastvisite) et de mettre le reste du code dans une boucle tant que tu as des utilisateurs à supprimer ... il te faudra donc faire un petit calcul avec les date unix pour dégager ton critère (tes deux ans sans connction ...)
c'est la solution la plus efficace je pense ...

**Comment by jipexu** [comment]
ah j'avais pas compris donc maintenant il te reste que les deux tables annexes à épurer ? là tu n'as pas de question à te poser tout les uid qui ne sont pas dans la table user tu vire.. en espérant que tu ai supprimer tout les reste comme dans case 'delUserConf': dans admin/users.php ....

**Comment by Jireck-npds** [comment]
je regardes des que j'ai un temps

**Comment by Jireck-npds** [comment]
je reponds avant d'avoir tester.
 il serait bien d'avoir la date de dernière connexion

**Comment by jipexu** [comment]
où ça ? dans quelle page ?? 
dans le tableau des utilisateurs en attente de validation? ça n'a guère de sens vu qu'ils ne se sont jamais connectés ...

**Comment by Jireck-npds** [comment]
je reprends le sujet et je reponds en vrac juste a la derniere remarque.
si il s'est inscrit et connecter depuis deux an mais que l'an dernier ne l'autorise plus (coche) il serait bien de le savoir non ?

**Comment by jipexu** [comment]
oui dans ton cas ... en fait quand on a écrit ça c'était pour les nouveaux utilisateurs et oui la deuxième date (dernière connexion) aurait du sens pour ceux que l'on a déconnecté après l'inscription ...
pour rappel ces infos sont aussi dispo dans memberlist ou tu peux directement supprimer ou editer un utilisateur aussi ...

**Comment by Jireck-npds** [comment]
En fait, je ne fais pas la difference entre les users en attente de validation, que la suppression ou edition d'un user.
Je prends le contexte global : 
J'ai des users à gerer que ce soit : la création, la modification / edition ou la suppression.
Pour moi, un outil "administrateur" doit pouvoir gérer tous les états du cycle de vie d'un utilisateurs et les informations sur eux.

Par exemple, tu permet la création d'utilisateur avec un minisite mais valider par l'admin.
l'admin doit pouvoir : 
- valider en masse les utilisateurs.
- Valider unitairement un utilisateur avec permission de se connecter et d'avoir ou non un mini-site..

bref tu m'as compris. ... ou pas

**Comment by jipexu** [comment]
oui ... il manque depuis toujours des fonctions pour la gestion (de masse) des utilisateurs

**Comment by Jireck-npds** [comment]
pour aider j'ai fait tourné ces 2 requetes : 
DELETE FROM users_extend WHERE uid NOT IN (SELECT uid  FROM users);
DELETE FROM users_status WHERE uid NOT IN (SELECT uid  FROM users);

on pourrait imaginer un function qui check  et delete si besoin les tables users_extends et users_status.
Genre : 
Select uid from  users_extend where uid NOT IN (SELECT uid  FROM users)
affichage difference
validation
delete

**Comment by Jireck-npds** [comment]
d'ailleurs, quand on clique sur les utilisateurs en attente on obtiens ca : 
<img width="1244" alt="image" src="https://github.com/npds/npds_dune/assets/16939826/a8be8439-5077-49e4-b58a-801322af145a">
mais on a pas d'obtions pour supprimer un à un les utilisateurs, et quand on fais la seul option possible, on n'est que dans l'edition de l'utilisateur ...

**Comment by jipexu** [comment]
oui c'est voulu qu' il n'y ai pas de lien direct de suppression ce n'est pas l'objet de cette interface qui est utilisateur en attente de validation...la moindre des chose c'est que tu vois le profil avant de l'autoriser ou pas à la connection ....alors on va la ou il y a les infos et donc la "commande" pour l'autorisation à la connection. et si tu veux le supprimer on  on retourne dans user pour suppression. 
tu as ce type d'interface que tu demande dans memberlist.php en admin ...

**Comment by Jireck-npds** [comment]
c'est dommage car c'est pas des utilisateurs en attente c'est des utilisateurs qui ne peuvent pas se connecter.
comme tu le vois dans mon impression écran c'est des vieux comptes qu'on a désactivé pour une raison ou une autre.

en fait l'outil ne permet pas de savoir si c'est (comme dit) une validation de connexion ou une reprise d'activité ou une suspension de compte.

puis le fichier memberlist est un fichier "user", il est dommage de devoir aller dans un fichier user en admin pour faire cette op.

**Comment by jipexu** [comment]
... Utilisateur(s) en attente de validation ... (peu importe la raison ...) cette interface est juste la pour répondre à l'alerte qui te prévient ....

oui des fichiers "user" qui ont des fonctions d'admin c'est loin d'être le seul ... (on pourrait mettre une commande dans l'admin ....? dans ce cas là) en tout cas dans ce cas là il est bien utile vu que cette liste n'existe pas dans l'admin

**Reply by Jireck-npds** [reply]
Bon.... oui elle est utile cette liste; 
je dis juste que c'est pas le bon terme : les utilisateurs sont en attente d'autorisation de connexion
je dis juste aussi qu'il est dommage d'aller dans un fichier qui fait pas parti de l'admin pour supprimer des comptes etc... 
Apres voila 
Vous faites comme vous le voulez

Je verrais à un moment a p'tre faire une "Gestion des utilisateurs" entre guillemet correct (pas de remise en cause du boulot déjà fait, c'est genial super etc ... ) parce l'outil ne ME convient pas, mais surement au autres.

**Comment by jipexu** [comment]
oui faire une liste de tous les utilisateurs avec les infos qui vont bien et les commandes nécessaire à un traitement de masse .....
